extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -1502,6 +1502,98 @@ describe('amp-a4a', () => {
     // FAILURE cases here
   });
 
+  describe('SRA Ads Delay Measure', () => {
+    let fixture;
+    beforeEach(() => {
+      xhrMock.withArgs(TEST_URL, {
+        mode: 'cors',
+        method: 'GET',
+        credentials: 'include',
+      }).onFirstCall().returns(Promise.resolve(mockResponse));
+      return createIframePromise().then(f => {
+        setupForAdTesting(f);
+        fixture = f;
+        return fixture;
+      });
+    });
+    it('sends as expected', () => {
+      const win = fixture.win;
+      const a4a = new MockA4AImpl(createA4aElement(fixture.doc, win));
+      a4a.buildCallback();
+      const emitLifecycleEventSpy =
+        sandbox.spy(a4a, 'protectedEmitLifecycleEvent_');
+      a4a.onLayoutMeasure();
+      expect(a4a.adPromise_).to.be.ok;
+      return a4a.adPromise_.then(() => {
+        expect(win['a4a-measuring-ad-urls-adsense']).to.be.ok;
+        expect(win['a4a-measuring-ad-urls-doubleclick']).to.not.be.ok;
+        return win['a4a-measuring-ad-urls-adsense'].then(() => {
+          // TODO: get resources to include element so reported object can
+          // be verified.
+          expect(emitLifecycleEventSpy.withArgs(
+            'sraBuildRequestDelay',
+            {
+              'met.delta.AD_SLOT_ID': sinon.match.number,
+              'totalSlotCount.AD_SLOT_ID': 0,
+              'totalUrlCount.AD_SLOT_ID': 0,
+              'type.AD_SLOT_ID': 'adsense',
+              'totalQueriedSlotCount.AD_SLOT_ID': 0,
+            })).to.be.calledOnce;
+        });
+      });
+    });
+    it('does not execute if already present', () => {
+      const win = fixture.win;
+      win['a4a-measuring-ad-urls-adsense'] = true;
+      const a4aElement = createA4aElement(fixture.doc, win);
+      const a4a = new MockA4AImpl(a4aElement);
+      a4a.buildCallback();
+      const emitLifecycleEventSpy =
+        sandbox.spy(a4a, 'protectedEmitLifecycleEvent_');
+      a4a.onLayoutMeasure();
+      expect(a4a.adPromise_).to.be.ok;
+      return a4a.adPromise_.then(() => {
+        expect(emitLifecycleEventSpy).to.not.be.calledWith(
+          'sraBuildRequestDelay', sinon.match.object);
+      });
+    });
+    it('creates separate executions for doubleclick vs adsense', () => {
+      const win = fixture.win;
+      const adsenseA4a = new MockA4AImpl(createA4aElement(fixture.doc, win));
```

Comment:
That is the default:

function createA4aElement(doc, opt_rect) {
    const element = createElementWithAttributes(doc, 'amp-a4a', {
      'width': opt_rect ? String(opt_rect.width) : '200',
      'height': opt_rect ? String(opt_rect.height) : '50',
      'type': 'adsense',
    });